### ISSUE TITLE
Duplicate In Manifest WebVTT Captions

### ISSUE THREAD

**Issue by triwav** [videojs/videojs-contrib-hls#1185]
Hello everyone. First want to say thanks for all the work on getting WebVTT captions added to this library. I am seeing an issue with captions showing up multiple times in the player. I'm also seeing this same issue in JW Player however Safari seems to handle things correctly. Below is a link to a sample: http://brianleighty.com/videojs-hls-test (ignore the hard coded captions at the top) and here's a link to the m3u8 itself that you can view in Safari https://www.dvidshub.net/webcast/11812.m3u8?dvrTime=1499094660
I'm running 5.7.0 of videojs-contrib-hls and am currently running 5.20.1 of videojs. I can try 6.X but I don't think that should make a difference. In testing different streams, it seems the issue lies with the fact that we have captions spanning multiple segments. My work around that I haven't tested yet is to make captions not span multiple segments but just have it start at the same time in the next segment but thought I would report this as an issue as this should work but isn't. Let me know if you need any more details. Thanks.

**Comment by gesinger** [comment]
Hey @triwav , thanks for reporting and for providing the example. I think it's doubling up because there are multiple instances of each caption. For instance, the first caption segment has 

```
WEBVTT
X-TIMESTAMP-MAP=MPEGTS:900000,LOCAL:00:00:10.000

01:24.360 --> 01:28.381
.
>> I THINK IT IS IN DIRECT
```

while the second has

```
WEBVTT
X-TIMESTAMP-MAP=MPEGTS:900000,LOCAL:00:00:10.000

01:24.360 --> 01:28.381
.
>> I THINK IT IS IN DIRECT

01:28.381 --> 01:34.531
CONTRADICTION TO WHAT WAS SAID
LAST WEEK, THAT HE DOESN'T
```

Notice that "I THINK IT IS IN DIRECT" is specified multiple times (as you said, because it crosses segment boundaries). We don't dedupe captions, as we just read them as if they are one big document that has been cut up into segments. That is probably the reason they are doubling up in the stream.

**Comment by triwav** [comment]
Gotcha. Are there any plans to change this? Like I said I can work around it but that sounds like it could be handled better.

**Comment by gesinger** [comment]
I think you're right @triwav . The spec reads:

"Each WebVTT Segment MUST contain all subtitle cues that are intended to be displayed during the period indicated by the segment EXTINF duration."

This means that if a subtitle is split over two segments, it should appear in both webvtt segments (by my interpretation). So we should handle the case presented here.

I'm going to consider this a bug unless anyone has other thoughts on it (or different interpretations). And thanks again for reporting @triwav

**Comment by mjneil** [comment]
Since the spec states it MUST contain all the cues for that time period, it may be as simple as removing all cues from the track that are between the start of the first cue in the segment and the end of the last cue in the segment. If there is overlap across segments, we will just be removing and immediately re-adding a the same cue.

**Comment by mjneil** [comment]
A fix for this has been released in v5.8.3. Please open a new issue if you continue to experience any issues.
